core/src/test/scala/org/apache/spark/scheduler/DAGSchedulerSuite.scala

```diff
@@ -2627,6 +2632,81 @@ class DAGSchedulerSuite extends SparkFunSuite with LocalSparkContext with TimeLi
     assert(countSubmittedMapStageAttempts() === 2)
   }
 
+  test("SPARK-23207: retry all the succeeding stages when the map stage is random") {
```

Comment:
We need to add a test where final result stage has not started (yet) but intermediate stages are retried on failure.
This will validate that if result stage has not yet started, it wont fail the job

Reply:
In this test, the result stage has not yet started.

I'll add a new test when the result stage is started.